From 53537d4af52e3afa78f5fc11411b1e757be4b33b Mon Sep 17 00:00:00 2001
From: Alexander Afanasyev <aa@cs.fiu.edu>
Date: Sat, 30 May 2020 18:09:32 -0400
Subject: [PATCH 5/5] security: Create PIB implementation as shared_ptr

While existing code is valid, it caused strange crashing issues on
Android/arm plantform.

Change-Id: I29c4879c65b3c844abb2d75f267c44a0b959a6f7
---
 ndn-cxx/security/v2/key-chain.hpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/ndn-cxx/security/v2/key-chain.hpp b/ndn-cxx/security/v2/key-chain.hpp
index 1021e8b5..369f4296 100644
--- a/ndn-cxx/security/v2/key-chain.hpp
+++ b/ndn-cxx/security/v2/key-chain.hpp
@@ -369,7 +369,7 @@ public: // PIB & TPM backend registry
   registerTpmBackend(const std::string& scheme);
 
 private:
-  typedef std::map<std::string, function<unique_ptr<pib::PibImpl>(const std::string& location)>> PibFactories;
+  typedef std::map<std::string, function<std::shared_ptr<pib::PibImpl>(const std::string& location)>> PibFactories;
   typedef std::map<std::string, function<unique_ptr<tpm::BackEnd>(const std::string& location)>> TpmFactories;
 
   static PibFactories&
@@ -456,7 +456,7 @@ inline void
 KeyChain::registerPibBackend(const std::string& scheme)
 {
   getPibFactories().emplace(scheme, [] (const std::string& locator) {
-      return unique_ptr<pib::PibImpl>(new PibType(locator));
+      return std::shared_ptr<pib::PibImpl>(new PibType(locator));
     });
 }
 
-- 
2.26.2

